### PR TITLE
FIX(voucher totals): Retotal voucher on recovery

### LINKED ISSUE #816: Journal Voucher : No total when reloading the form
When reloading the page to continue the transcation adding process, there is no value for total.
**Step to bug** 
1. Go to journal voucher
2. make a transaction
3. Before validate leave the page and go somewhere else
4. Return to the journal voucher page
5. Click the button to reload the previous form.

**Notice**
 No value for the sum.

### PR DESCRIPTION
**Pull request by jniles**
This commit adds the ability to retotal a journal voucher when the
voucher loaded from stored data.

I've also updated the controller to use $inject syntax while I was
there.  Notably, `self` --> `vm`, to standardize our controllers.

Fixes #816.

### CODE REVIEW

File: client/src/partials/journal_voucher/journal_voucher.js [hunk 1]
```diff
@@ -1,238 +1,246 @@
...
+    rows : [] // the child
+  };
 
-    // toggle reference field
-    self.toggleReference = function () {
-      self.showReference = !self.showReference;
-    };
+  // load dependencies
+  $http.get('/finance/currencies')
+  .success(function (data) {
```
Comment (DedrickEnc) [inline client/src/partials/journal_voucher/journal_voucher.js]: ```success``` is deprecated as discussed

Comment (jniles) [inline client/src/partials/journal_voucher/journal_voucher.js]: This is now fixed.

File: client/src/partials/journal_voucher/journal_voucher.js [hunk 2]
```diff
@@ -277,35 +289,35 @@ angular.module('bhima.controllers')
...
-  .success(function (data) {
-    self.accounts = data;
+  .then(function (response) {
+    vm.accounts = response.data;
   })
-  .error(handle);
+  .catch(handle);
 
   // load debtors
   $http.get('/finance/debtors')
```
Comment (DedrickEnc) [inline client/src/partials/journal_voucher/journal_voucher.js]: you should use ```.then``` instead of ```.success```.

Comment (jniles) [inline client/src/partials/journal_voucher/journal_voucher.js]: Hm.  You should refresh your page. :)

File: client/src/partials/journal_voucher/journal_voucher.js [hunk 3]
```diff
@@ -370,10 +382,9 @@ angular.module('bhima.controllers')
   // set the account for a row
   // remove all debtor/creditor properties that may
   // have been set on a previous selection
-  self.setAccount = function (row) {
+  vm.setAccount = function (row) {
     row.account_id = row.account.id;
     row.deb_cred_uuid = undefined;
```
Comment (DedrickEnc) [inline client/src/partials/journal_voucher/journal_voucher.js]: ```undefined``` means the variable is declared but no value is assigned yet, to be semantically correct you should assign ```null``` which is an assignment value, and it is also a representation of no value.

**Comment by jniles** [comment]
@dedrickenc, all the fixes have landed.

**Pull request merged**